convex/analytics.ts

```
@@ -0,0 +1,253 @@
+    >()
+
+    filteredUsage.forEach(record => {
+      const date = new Date(record.timestamp)
+      let key: string
+
+      switch (args.interval) {
+        case 'day':
+          key = date.toISOString().split('T')[0] // YYYY-MM-DD
+          break
+        case 'week':
+          const weekStart = new Date(date)
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)